contracts/VetoMultisigVoting.sol

```diff
@@ -0,0 +1,243 @@
+
+            emit FreezeProposalCreated(msg.sender);
+        } else {
+            // There is an existing freeze proposal, count user's votes
+            require(
+                !userHasFreezeVoted[msg.sender][freezeProposalCreatedBlock],
+                "User has already voted"
+            );
+
+            freezeProposalVoteCount++;
+        }
+
+        userHasFreezeVoted[msg.sender][freezeProposalCreatedBlock] = true;
+
+        emit FreezeVoteCast(msg.sender, 1);
+    }
+
+    /// @notice Unfreezes the DAO, only callable by the owner
+    function defrost() public onlyOwner {
+        require(isFrozen(), "DAO is not already frozen");
+        freezeProposalCreatedBlock = 0;
+        freezeProposalVoteCount = 0;
+    }
+
+    /// @notice Updates the veto votes threshold, only callable by the owner
+    /// @param _vetoVotesThreshold The number of votes required to veto a transaction
+    function updateVetoVotesThreshold(uint256 _vetoVotesThreshold)
+        external
+        onlyOwner
+    {
+        vetoVotesThreshold = _vetoVotesThreshold;
+    }
+
+    /// @notice Updates the freeze votes threshold, only callable by the owner
+    /// @param _freezeVotesThreshold Number of freeze votes required to activate a freeze
+    function updateFreezeVotesThreshold(uint256 _freezeVotesThreshold)
+        external
+        onlyOwner
+    {
+        freezeVotesThreshold = _freezeVotesThreshold;
+    }
+
+    /// @notice Updates the freeze proposal blocks duration, only callable by the owner
+    /// @param _freezeProposalBlockDuration The number of blocks a freeze proposal has to succeed
+    function updateFreezeProposalBlockDuration(
+        uint256 _freezeProposalBlockDuration
+    ) external onlyOwner {
+        freezeProposalBlockDuration = _freezeProposalBlockDuration;
+    }
+
+    /// @notice Updates the freeze block duration, only callable by the owner
+    /// @param _freezeBlockDuration The number of blocks a freeze last, from time of freeze proposal creation
+    function updateFreezeBlockDuration(uint256 _freezeBlockDuration)
+        external
+        onlyOwner
+    {
+        freezeBlockDuration = _freezeBlockDuration;
+    }
+
+    /// @notice Returns whether the specified transaction has been vetoed
+    /// @param to Destination address.
+    /// @param value Ether value.
+    /// @param data Data payload.
+    /// @param operation Operation type.
+    /// @param safeTxGas Gas that should be used for the safe transaction.
+    /// @param baseGas Gas costs for that are independent of the transaction execution(e.g. base transaction fee, signature check, payment of the refund)
+    /// @param gasPrice Maximum gas price that should be used for this transaction.
+    /// @param gasToken Token address (or 0 if ETH) that is used for the payment.
+    /// @param refundReceiver Address of receiver of gas payment (or 0 if tx.origin).
+    /// @return bool True if the transaction is vetoed
+    function getIsVetoed(
```

Comment:
I really like having `getIsVetoed` be a function on the `IVetoVoting` interface. This makes the contract interactions between the `VetoGuard` and this contract cleaner. `VetoGuard` just calls this one function on `VetoVoting`, instead of calling two separate functions and doing the check itself.